cora/signal/lss.py

```diff
@@ -703,6 +785,40 @@ def process(self, initial_field: InitialLSS, biased_field: BiasedLSS) -> BiasedL
         return final_field
 
 
+# List the models that are available for validation by the classes
+_omega_HI_models = ["Crighton2015", "SKA", "uniform"]
+
+
+def omega_HI(
+    z: FloatArrayLike, model: Literal["Crighton2015", "SKA", "uniform"] = "Crighton2015"
+):
+    """Calculate the HI fraction at specified redshifts.
+
+    Parameters
+    ----------
+    z
+        The redshifts to calculate at.
+    model
+        The model to use for Omega_HI. See `Notes` for details.
+
+    Notes
+    -----
+    We use two models for the evolution of Omega_HI. A fit from Crighton et al. 2015
+    [1]_ (`Crighton2015`) and a fit from the SKA WG whitepaper [2]_ (`SKA`) as well as
+    allowing a constant in redshift value to be set (`uniform`), with a value
+    taken from Switzer et al. 2013 [3]_ assuming :math:`b_{HI}=1`.
+    """
+
+    if model == "uniform":
+        return np.ones_like(z) * 0.6e-3
```

Comment:
I think we should retain the option for the user to specify a constant OmegaHI value (both in this routine and in the relevant tasks), rather than having it hard-coded

Reply:
I kind of agree, but I couldn't think of a good and quick way of doing it without a lot of repetition. In theory people can just use the `map_prefactor` option to achieve the same thing. I'm going to add a TODO and pass on this for now as it's better to have something run through.

Reply:
I guess really what we need is a way to configure the Omega_HI model globally, but I'm not sure there's actually a nice way to do that one without some fundamental changes to the pipeline framework.